src/pyproject_hooks/_impl.py

```diff
@@ -260,10 +317,16 @@ def prepare_metadata_for_build_editable(
         )
 
     def build_editable(
-        self, wheel_directory, config_settings=None, metadata_directory=None
-    ):
+        self,
+        wheel_directory: str,
+        config_settings: Optional[Mapping[str, Any]] = None,
```

Comment:
```suggestion
        config_settings: Mapping[str, Any] | None = None,
```

The `_directory` params should use StrPath.